File: 백준/Gold/6198. 옥상 정원 꾸미기/옥상 정원 꾸미기.js
const path = process.platform === 'linux' ? '/dev/stdin' : './input.txt';
let input = require('fs').readFileSync(path).toString().trim().split(/\s+/).map(Number);

const [N, ...buildings] = input.map(Number);

const stack = [];
let result = 0;

for (let i = 0; i < N; i++) {
  let now = buildings[i];
  while (stack.length > 0 && stack[stack.length - 1] <= now) {
    stack.pop();
  }
  stack.push(now);
  result += stack.length - 1;
}
console.log(result);

File: 백준/Gold/6198. 옥상 정원 꾸미기/옥상 정원 꾸미기.py
import sys
input = sys.stdin.readline

N = int(input())
stack = []
result = 0 

concierges = [int(input()) for _ in range(N)]
for i in range(N):
  cur_concierge = concierges[i]

  while stack and cur_concierge >= stack[-1]:
    stack.pop()
      

  result += len(stack)
  stack.append(cur_concierge)

print(result)